frappe/core/doctype/user/user.py

```diff
@@ -98,7 +99,6 @@ def on_update(self):
 		clear_notifications(user=self.name)
 		frappe.clear_cache(user=self.name)
 		self.send_password_notification(self.__new_password)
-		create_contact(self, ignore_mandatory=True)
```

Comment:
@shridarpatil  If User email is changed, then there wont be any contact that exists with the new email id.

Reply:
Ah I missed that. Can we move this to background job?

Reply:
- We could move it to background jobs and get away with it
- If we do not create contact while creating user, and if welcome email is sent, the system will create a contact in communication [here](https://github.com/frappe/frappe/blob/2b2794d675fbd8963eb92c3a2fc3521bd8b71d64/frappe/core/doctype/communication/communication.py#L360). Then when background job is run, we can check if email exists in the system, then update the relevant info like name, phone as per user, if contact doesn't exist, we can create one.